cloud_function/database-migration/migrate_database.py

```
@@ -390,6 +395,67 @@ def create_table_structure(source_conn, dest_conn, table_name):
+    dest_cursor = dest_conn.cursor()
+
+    try:
+        # Get source columns
+        source_cursor.execute(f"""
+            SELECT column_name, data_type, character_maximum_length,
+                   numeric_precision, numeric_scale, is_nullable, column_default
+            FROM information_schema.columns
+            WHERE table_schema = 'public'
+            AND table_name = '{table_name}'
+            ORDER BY ordinal_position;
+        """)
```

Ruff (0.14.5)

Possible SQL injection vector through string-based query construction

(S608)